### PR TITLE
chore(suite): refactor coinmarket sections

### LINKED ISSUE #14852: Trading - long amount or symbols does not fit to UI nicely
**Describe the bug**

Version: 24.10.1

Screenshot 2024-10-14 at 19 46 16
Screenshot 2024-10-14 at 19 46 43
Screenshot 2024-10-14 at 19 44 10
Screenshot 2024-10-14 at 19 44 59

### PR DESCRIPTION
**Pull request by seibei-iguchi**
### Before
Screenshot 2024-10-31 at 11 38 51
Screenshot 2024-10-31 at 11 38 31

### After
Screenshot 2024-10-31 at 11 34 37
Screenshot 2024-10-31 at 11 35 17

### CODE REVIEW

File: packages/components/src/components/Card/Card.tsx [hunk 1]
```diff
@@ -25,10 +25,11 @@ export const allowedCardFrameProps = [
 ] as const satisfies FramePropsKeys[];
 type AllowedFrameProps = Pick<FrameProps, (typeof allowedCardFrameProps)[number]>;
 
-const Container = styled.div<TransientProps<AllowedFrameProps>>`
+const Container = styled.div<{ $fillType: FillType } & TransientProps<AllowedFrameProps>>`
```
Comment (peter-sanderson) [inline packages/components/src/components/Card/Card.tsx:28]: nit: export to type, sometimes this does issues on VSCode, but its not a big deal

File: packages/components/src/components/form/SelectBar/SelectBar.tsx [hunk 2]
```diff
@@ -35,14 +44,18 @@ const getTranslateValue = (index: number) => {
 const getPuckWidth = (optionsCount: number) =>
     `calc((100% - 8px - ${(optionsCount - 1) * spacings.xxs}px) / ${optionsCount})`;
 
-const Options = styled.div<{ $optionsCount: number; $isFullWidth?: boolean }>`
+const Options = styled.div<{
```
Comment (peter-sanderson) [inline packages/components/src/components/form/SelectBar/SelectBar.tsx:47]: nit: export type

File: packages/components/src/components/form/SelectBar/SelectBar.tsx [hunk 3]
```diff
@@ -53,16 +66,16 @@ const Options = styled.div<{ $optionsCount: number; $isFullWidth?: boolean }>`
     }
 `;
 
-const Puck = styled.div<{ $optionsCount: number; $selectedIndex: number }>`
+const Puck = styled.div<{ $optionsCount: number; $selectedIndex: number; $elevation: Elevation }>`
```
Comment (peter-sanderson) [inline packages/components/src/components/form/SelectBar/SelectBar.tsx:69]: nit: export type

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketForm/CoinmarketFormInput/CoinmarketFormInputCurrency.tsx [hunk 4]
```diff
@@ -22,14 +22,6 @@ import {
 } from 'src/utils/wallet/coinmarket/coinmarketTypingUtils';
 import { buildFiatOption } from 'src/utils/wallet/coinmarket/coinmarketUtils';
 import { CoinmarketFormOption, CoinmarketFormOptionLabel } from 'src/views/wallet/coinmarket';
-import styled from 'styled-components';
-
-const SelectWrapper = styled.div`
-    /* stylelint-disable selector-class-pattern */
-    .react-select__value-container {
```
Comment (peter-sanderson) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketForm/CoinmarketFormInput/CoinmarketFormInputCurrency.tsx:29]: :clap:  :clap:  :clap: 
 
image

**Review by peter-sanderson — APPROVED** [review]
:clap: good job

**Comment by enjojoy** [comment]
nit: I would add a line break after the first sentence 
image

**Review by enjojoy — APPROVED** [review]
Nice, so much custom styles eliminated

**Review by adderpositive — CHANGES_REQUESTED** [review]
Some details.

1. Is this expected? 

obrazek

---

2. Can you unify these two labels? And also fix the component's background in dark mode?

obrazek
obrazek
obrazek

---

3. Fix the buy test please.

---

The slippage component is very nice! I love it. Thank you for the improvements.

**Comment by seibei-iguchi** [comment]
> Some details.
> 
> 
> 
> 1. Is this expected? 
> 
> 
> 
> obrazek
> 
> 
> 
> ---
> 
> 
> 
> 2. Can you unify these two labels? And also fix the component's background in dark mode?
> 
> 
> 
> obrazek
> 
> obrazek
> 
> obrazek
> 
> 
> 
> ---
> 
> 
> 
> 3. Fix the buy test please.
> 
> 
> 
> ---
> 
> 
> 
> The slippage component is very nice! I love it. Thank you for the improvements. 
> 
> 

You're right about the 2. I'm not sure about the 1, it's part of the issue this PR is linked to. @tomasklim what do you think?

**Comment by tomasklim** [comment]
> @tomasklim what do you think?

@adamhavel 
How about allowing it to go to the next line? And also please add `isBalance={false}` to component with `data-testid` equal to `@coinmarket/offers/quote/crypto-amount`. That will remove `...` which does not make any sense here. The `formatCoinBalance` method is something random 😀 
Screenshot 2024-10-31 at 15 48 21

**Comment by tomasklim** [comment]
No sorry, it is `CoinmarketFormOfferCryptoAmount`, just remove `isBalance` there. And add `ellipsisLineCount={2}` to `Text`

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketForm/CoinmarketFormOffer.tsx [hunk 5]
```diff
@@ -115,93 +73,89 @@ export const CoinmarketFormOffer = () => {
...
-            )}
+                    <TextButton
+                        onClick={async () => {
+                            setIsCompareLoading(true);
+                            await goToOffers();
+                        }}
+                        size="small"
+                        isDisabled={state.isLoadingOrInvalid}
+                        isLoading={isCompareLoading}
+                        data-testid="@coinmarket/form/compare-button"
```
Comment (adderpositive) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketForm/CoinmarketFormOffer.tsx:118]: Adding type="button" will fix the test

**Review by adderpositive — APPROVED** [review]

**Pull request merged**